packages/flutter_tools/lib/src/flutter_plugins.dart

```diff
@@ -1137,127 +1137,156 @@ List<PluginInterfaceResolution> resolvePlatformImplementation(
     MacOSPlugin.kConfigKey,
     WindowsPlugin.kConfigKey,
   ];
-  final List<PluginInterfaceResolution> finalResolution = <PluginInterfaceResolution>[];
+  final List<PluginInterfaceResolution> pluginResolutions = <PluginInterfaceResolution>[];
   bool hasResolutionError = false;
 
   for (final String platformKey in platformKeys) {
     // Key: the plugin name
-    final Map<String, List<Plugin>> possibleResolutions = <String, List<Plugin>>{};
+    final Map<String, List<Plugin>> pluginImplCandidates = <String, List<Plugin>>{};
     final Map<String, String> defaultImplementations = <String, String>{};
 
     for (final Plugin plugin in plugins) {
-      final String? defaultImplementation = plugin.defaultPackagePlatforms[platformKey];
-      if (plugin.platforms[platformKey] == null && defaultImplementation == null) {
-        // The plugin doesn't implement this platform.
-        continue;
-      }
-      String? implementsPackage = plugin.implementsPackage;
-      if (implementsPackage == null || implementsPackage.isEmpty) {
-        final bool hasInlineDartImplementation =
-            plugin.pluginDartClassPlatforms[platformKey] != null;
```

Comment:
Here an "inline dart implementation" is defined when if it has no dart plugin class. [Here](https://github.com/flutter/flutter/pull/145258/files#diff-68361f6fc9aafa5fa7de042eec4603dd4f321eb783a00edd57780ef78c1094bbL1193) also it is excluded, if it has the `none` keyword. I suspect the `none` String basically is never used so it worked, but I unified the variable [here](https://github.com/flutter/flutter/pull/145258/files#diff-68361f6fc9aafa5fa7de042eec4603dd4f321eb783a00edd57780ef78c1094bbR1203). If you have doubts, let me know.